Doc/Manual/Python.html

```diff
@@ -4971,6 +4971,60 @@ <H3><a name="Python_nn59">33.9.1 Converting  Python list to a char ** </a></H3>
 </pre></div>
 
 <p>
+  The following SWIG interface file allows a Python3 list
+  object to be used as a <tt>char **</tt> object.
+</p>
+
+
+<div class="code"><pre>
+  %module argv
+
+  // This tells SWIG to treat char ** as a special case
+  %typemap(in) char ** {
+    /* Check if is a list */
+    if (PyList_Check($input)) {
+      int size = PyList_Size($input);
+      int i = 0;
+      $1 = (char **) malloc((size+1)*sizeof(char *));
+      for (i = 0; i < size; i++) {
+        PyObject *o = PyList_GetItem($input, i);
+        if (PyUnicode_Check(o)) {
+          $1[i] = strdup(PyUnicode_AsUTF8(o));
+        } else {
+          free($1);
+          PyErr_SetString(PyExc_TypeError, "list must contain strings");
+          SWIG_fail;
+        }
+      }
+      $1[i] = 0;
+    } else {
+      PyErr_SetString(PyExc_TypeError, "not a list");
+      SWIG_fail;
+    }
+  }
+
+  // This cleans up the char ** array we malloc'd before the function call
+  %typemap(freearg) char ** {
+    free((char *) $1);
```

Comment:
This frees the `char **` array but leaks all the allocations in it made by calling `strdup()`.